openedx/core/djangoapps/courseware_api/views.py

```diff
@@ -58,11 +58,31 @@ def __init__(self, course_key, request, username=''):
             course_key,
             staff_access=self.is_staff,
         )
-        self.course_masquerade = course_masquerade
+        self._course_masquerade = course_masquerade
 
     def __getattr__(self, name):
         return getattr(self.overview, name)
 
+    @property
+    def course_masquerade(self):
+        """
+        Return details about current masquerade status
+        """
+        course = self._course_masquerade
+        course_key_string = str(self.course_key)
+        data = {}
+        if course:
+            data = {
+                'active': {
+                    'course_key': course_key_string,
+                    'group_id': course.group_id,
+                    'role': course.role,
+                    'user_name': course.user_name or ' ',
```

Comment:
TODO: This may coalesce to an empty string, not a space.

I think this was originally chosen to maintain parity with some existing code that did this, but I _think_ we not longer need to preserve that weird back-compat.

I'll update with conclusions prior to hand-off.

Reply:
Done.